src/lumigo_tracer/event/event_trigger.py

```diff
@@ -197,6 +197,10 @@ def _parse_streams(event: dict) -> Dict[str, str]:
         result.update(_parse_sqs_event(event))
     elif triggered_by == "kinesis":
         result["messageId"] = safe_get(event, ["Records", 0, "kinesis", "sequenceNumber"])
+        result["recordsNum"] = len(safe_get(event, ["Records"]))
```

Comment:
to spare performance, you can do
```
records = safe_get(event, ["Records"])
```

Reply:
Not sure I understand what is the improvement here?

Reply:
checkout lines 199-201 - they all do the same thing

Reply:
I don't follow you, all those safe_get are o(1) complexity no ?

Reply:
yes.
But they are wrapped with many `try/except` which may cause performance issues on the edgecase of many small requests.
Well. It is not that important. I'm taking it back. 👍